src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -888,10 +941,15 @@ private Type EmitBinaryExpressionNode(AssociativeNode node)
                         // variable being assigned already exists in dictionary.
                         throw new Exception("Variable redefinition is not allowed.");
                     }
-                    variables.Add(lNode.Value, new Tuple<int, Type>(++localVarIndex, t));
+                    variables.Add(lNode.Value, new Tuple<int, Type>(-1, t));
```

Comment:
is this change made because you want to always use the index defined by the local builder?

Reply:
We still need to cache the local var index in this dictionary as we need to query it for the index in two places: in `EmitArrayCoercion` and in `EmitIEnumerableCoercion`.